entgql/plugin/plugin.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"fmt"
+	"strings"
+
+	"entgo.io/contrib/entgql"
+	"entgo.io/ent/entc/gen"
+	"entgo.io/ent/schema/field"
+	"github.com/99designs/gqlgen/plugin"
+	"github.com/vektah/gqlparser/v2/ast"
+	"github.com/vektah/gqlparser/v2/formatter"
+)
+
+type (
+	// EntGQL is a plugin that generates GQL schema from the Ent's Graph
+	EntGQL struct {
+		graph  *gen.Graph
+		nodes  []*gen.Type
+		schema *ast.Schema
+
+		relaySpec bool
+	}
+
+	// EntGQLPluginOption is a option for the EntGQL plugin
+	EntGQLPluginOption func(*EntGQL) error
```

Comment:
```suggestion
	EntGQLOption func(*EntGQL) error
```

Avoid using the word `plugin` inside the package `plugin`. Redundant stuttering.